docs/knowledge-base/routing-sytem.md

```diff
@@ -0,0 +1,146 @@
+Such as being able to create routes on more than just the path, for example make a request for JSON, XML or HTML while still using the same path. 
+
+
+## How the routing system works
+
+Drupal's routing system works with the Symfony HTTP Kernel. 
+The routing system is responsible for matching paths to controllers, and you define those relations in routes. You can pass on additional information to your controllers in the route. Access checking is integrated as well.
+[See drupal.org][5]
+
+## Structure of routes and parameters
+
+The route structure inside the *module.routing.yml* file is described in [drupal.org][3]
+
+## Adding new routes (static & dynamic) and altering existing ones
+
+
+### Adding a new static route
+
+1. In your example.routing.yml (let's use **example** module name as in drupal.org)
+    
+    Add your path, ( /something )
+        and in *defaults:* 
+            add the controller class and the method who will do something.
+            set a title
+        as *requirements:*
+            add permission   
+        
+
+```
+example.content:
+  path: '/example' 
+  defaults: 
+    _controller: '\Drupal\example\Controller\ExampleController::content' 
+    _title: 'Hello World'
+  requirements: 
+    _permission: 'access content' 
+```
+    
+
+2. In your src/Controller add your page controller class and your methods
+
+
+As basic route controller you can use Drupal\Core\Controller\ControllerBase.
+
+```namespace Drupal\example\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+
+/**
+ * An example controller.
+ */
+class ExampleController extends ControllerBase {
+
+  /**
+   * Returns a render-able array for a test page.
+   */
+  public function content() {
+    $build = [
+      '#markup' => $this->t('Hello World!'),
+    ];
+    return $build;
+  }
+}
+```
+
+### Adding a new dynamic route
+
+You can use the alterRoutes() method to add dynamic routes as well. 
+A real example can be found in the [configuration translation RouteSubscriber][9].
+
+See [drupal.org][7]
+
+### Altering existing routes
+
+Altering existing routes is done by modifying a **RouteCollection** and using  an **EventSubscriber** triggered by the **RoutingEvents:ALTER** event.
+
+The example use a src/Routing/RouteSubscriber.php file in your module.
+Therefore, the class must be registered as an event subscriber service.
+Use a example.services.yml file in your module (if the module is named example).
+```
+services:
+  example.route_subscriber:
+    class: Drupal\example\Routing\RouteSubscriber
+    tags:
+      - { name: event_subscriber }
+```
+[See drupal.org][6]
+
+## Route related objects
+
+Route, CurrentRouteMatch, RouteMatch, Url are objects used for routing in Drupal 8.
+-RouteMatch: \Drupal\Core\Routing\RouteMatchInterface
+-CurrentRouteMatch: \Drupal\Core\Routing\CurrentRouteMatch
+
+see in [drupal.org][8]
+
+You can also  create your route into your module creation through [drupal console][10]:
```

Comment:
This doesn't belong to this section, it's not about objects related with routes.